### ISSUE TITLE
Security Requirements

### ISSUE THREAD

**Issue by cgye** [cds-snc/saas-procurement#329]

**Comment by dinophile** [comment]
SaaS procurement feature is confirmed to need PBMM controls. As per Jenn S.

**Comment by cgye** [comment]
Non-functional security & privacy requirements to consider for cloud-based GC services:

- Compliance with legal & regulatory requirements (e.g. PBMM security controls)
- Access control (i.e Authentication & Authorization)
  - Infrastructure
  - Application
- Accounting (i.e. Logging & Monitoring)

**Comment by cgye** [comment]
The following are *currently executed*, which are in scope, Security requirements per _Section 3.1 Mandatory Requirements_ of [Secure Software Development Lifecycle](https://docs.google.com/document/d/1U8_7zAeJN868yDWXILPcO-o6BTNlAQNdAQMUNpSFJnc/edit?usp=sharing):
- Security Categorization
   - Classify Information
   - Establish System Security Control Profile
- Auditing
   - Configure or use an authoritative time source for the time-stamp of the audit records generated by your solution components.
- Networking
   - Network boundary protection (e.g. WAF) for external facing interfaces (deny-all or allow-by-exception policy)
- Secure Development
   - Static Application Security Testing (SAST)
   - Dynamic Application Security Testing (DAST)
   - Penetration Testing
   - Vulnerability Scanning
- Security Operations
   - Automated Vulnerability Scanning
   - Vulnerability and Patch management process
   - Incident Response Plan
   - System Monitoring Process

The following are *new*, where the scope is up for discussion, Security requirements per _Section 3.1 Mandatory Requirements_ of [Secure Software Development Lifecycle](https://docs.google.com/document/d/1U8_7zAeJN868yDWXILPcO-o6BTNlAQNdAQMUNpSFJnc/edit?usp=sharing):
- System Concept
   - Business Requirements
   - Identify all user and service management roles
   - Describe all operational scenarios
- Identity and Access Management
   - Uniquely identify and authenticate users (no anonymize access)
   - Multi-factor authentication for privileged accounts
   - Process for managing access privileges (Principle of Least Privilege)
   - Enforce access authorizations
- Data Protection
   - Encryption of data in transit (Guidance on securely configuring network protocols ITSP.40.062)
   - Encryption of data at rest (Cryptographic algorithms for UNCLASSIFIED, PROTECTED A, and PROTECTED B Information - ITSP.40.111)
   - Secret management procedures
- Auditing
   - Identify the events within the solution that must be audited
   - Implement an audit process for the auditable events identified. At a minimum, the business owner or an auditor should audit actions against user accounts according to SPIN 2017‑01.
   - Protect audit information by controlling access to the audit log tools.
- Service Continuity
   - Backup-and-restore process for GC data
   - Fail-over and availability (e.g. for MM)
- Configuration Management
   - Define and document baseline configuration, which represents the most restrictive (secure) mode of operation
   - Implement a process to keep baseline configuration up to date as changes are implemented
- Security Operations
   - Periodic Penetration Testing
